crates/fixedpointmath/src/math.rs

```diff
@@ -0,0 +1,450 @@
+                        *self = self.[<$trait:lower>](other);
+                    }
+                }
+            }
+        )*
+    };
+    ($($tt:tt)*) => {};
+}
+
+mapped_operator_impls!(
+    // use `mul_down` for `*` and `*=`.
+    Mul => mul_down,
+    // use `div_down` for `/` and `/=`.
+    Div => div_down
+);
+
+/// Takes a list of operator traits and implements the operator and assignment
+/// operator for each one by forwarding to the corresponding method on the
+/// underlying `FixedPointValue`.
+macro_rules! forwarded_operator_impls {
+    ($($trait:ident),*) => {
+        $(
+            paste::paste! {
+
+                impl<T: FixedPointValue> std::ops::$trait for FixedPoint<T> {
+                    type Output = Self;
+
+                    fn [<$trait:lower>](self, other: Self) -> Self::Output {
+                        Self::new(self.raw().[<$trait:lower>](other.raw()))
+                    }
+                }
+
+                impl<T: FixedPointValue> std::ops::[<$trait Assign>] for FixedPoint<T> {
+                    fn [<$trait:lower _assign>](&mut self, other: Self) {
+                        *self = self.[<$trait:lower>](other);
+                    }
+                }
+            }
+        )*
+    };
+    ($($tt:tt)*) => {};
+}
+
+// Forward these operators to the underlying `FixedPointValue`.
+forwarded_operator_impls!(Add, Sub, Rem);
+
+#[cfg(test)]
+mod tests {
+    use std::{panic, u128};
+
+    use ethers::{signers::Signer, types::U256};
+    use eyre::Result;
+    use hyperdrive_wrappers::wrappers::mock_fixed_point_math::MockFixedPointMath;
+    use rand::{thread_rng, Rng};
+    use test_utils::{chain::Chain, constants::DEPLOYER};
+
+    use super::*;
+    use crate::{fixed, fixed_u128, fixed_u256, uint256};
+
+    /// The maximum number that can be divided by another in the Solidity
+    /// implementation.
+    fn max_sol_numerator() -> FixedPoint<U256> {
+        (U256::MAX / uint256!(1e18)).into()
+    }
+
+    #[test]
+    fn test_sub_failure() {
+        // Ensure that subtraction failures are propagated from the raw type.
+        assert!(panic::catch_unwind(|| fixed_u256!(1e18) - fixed!(2e18)).is_err());
+    }
+
+    #[test]
+    fn test_mul_div_down_failure() {
+        // Ensure that division by zero fails.
+        assert!(
+            panic::catch_unwind(|| fixed_u128!(1e18).mul_div_down(fixed!(1e18), fixed!(0)))
+                .is_err()
+        );
+    }
+
+    #[tokio::test]
+    async fn fuzz_mul_div_down() -> Result<()> {
+        let chain = Chain::connect(None, None).await?;
+        chain.deal(DEPLOYER.address(), uint256!(100_000e18)).await?;
+        let client = chain.client(DEPLOYER.clone()).await?;
+        let mock_fixed_point_math = MockFixedPointMath::deploy(client, ())?.send().await?;
+
+        // Fuzz the rust and solidity implementations against each other.
+        let mut rng = thread_rng();
+        for _ in 0..10_000 {
```

Comment:
can you use the FUZZ constant here & elsewhere?

Reply:
Oh, I remember why the `const` isn't used here. It comes from the `hyperdrive-test-utils` crate which isn't a dependency of `fixedpointmath` (because it already depends on `fixedpointmath` itself).

I'm going to leave these for now. But maybe we reconsider the env variable names for all constants across crates, so they can use their own const defs (without depending on eachother), but still share the same values by referencing the same env vars.